packages/twenty-server/src/engine/api/graphql/workspace-query-builder/factories/record-position-query.factory.ts

```diff
@@ -1,54 +1,119 @@
 import { Injectable } from '@nestjs/common';
 
+import { computeTableName } from 'src/engine/utils/compute-table-name.util';
+
 export enum RecordPositionQueryType {
-  GET = 'GET',
-  UPDATE = 'UPDATE',
+  FIND_MIN_POSITION = 'FIND_MIN_POSITION',
+  FIND_MAX_POSITION = 'FIND_MAX_POSITION',
+  FIND_BY_POSITION = 'FIND_BY_POSITION',
+  UPDATE_POSITION = 'UPDATE_POSITION',
 }
 
+type FindByPositionQueryArgs = {
+  positionValue: number;
+  recordPositionQueryType: RecordPositionQueryType.FIND_BY_POSITION;
+};
+
+type FindMinPositionQueryArgs = {
+  recordPositionQueryType: RecordPositionQueryType.FIND_MIN_POSITION;
+};
+
+type FindMaxPositionQueryArgs = {
+  recordPositionQueryType: RecordPositionQueryType.FIND_MAX_POSITION;
+};
+
+type UpdatePositionQueryArgs = {
+  recordId: string;
+  positionValue: number;
+  recordPositionQueryType: RecordPositionQueryType.UPDATE_POSITION;
+};
+
+type RecordPositionQuery = string;
+
+type RecordPositionQueryParams = any[];
+
+export type RecordPositionQueryArgs =
+  | FindByPositionQueryArgs
+  | FindMinPositionQueryArgs
+  | FindMaxPositionQueryArgs
+  | UpdatePositionQueryArgs;
+
 @Injectable()
 export class RecordPositionQueryFactory {
-  async create(
-    recordPositionQueryType: RecordPositionQueryType,
-    positionValue: 'first' | 'last' | number,
+  create(
+    recordPositionQueryArgs: RecordPositionQueryArgs,
     objectMetadata: { isCustom: boolean; nameSingular: string },
     dataSourceSchema: string,
-  ): Promise<string> {
-    const name =
-      (objectMetadata.isCustom ? '_' : '') + objectMetadata.nameSingular;
-
-    switch (recordPositionQueryType) {
-      case RecordPositionQueryType.GET:
-        if (typeof positionValue === 'number') {
-          throw new Error(
-            'RecordPositionQueryType.GET requires positionValue to be a number',
-          );
-        }
-
-        return this.createForGet(positionValue, name, dataSourceSchema);
-      case RecordPositionQueryType.UPDATE:
-        return this.createForUpdate(name, dataSourceSchema);
+  ): [RecordPositionQuery, RecordPositionQueryParams] {
+    const name = computeTableName(
+      objectMetadata.nameSingular,
+      objectMetadata.isCustom,
+    );
+
+    switch (recordPositionQueryArgs.recordPositionQueryType) {
+      case RecordPositionQueryType.FIND_BY_POSITION:
+        return this.buildFindByPositionQuery(
+          recordPositionQueryArgs as FindByPositionQueryArgs,
```

Comment:
I feel like your structure allows you to avoid type assertion here, maybe it would be better to use type guards instead of this switch?

Reply:
good point. Rather than guards, I switched for `satisfies` instead of `as`